src/openassetio-core/TraitsData.cpp

```diff
@@ -58,6 +58,18 @@ class TraitsData::Impl {
     data_[traitId][propertyKey] = std::move(propertyValue);
   }
 
+  [[nodiscard]] trait::TraitProperties traitProperties(const trait::TraitId& traitId) const {
+    const auto& traitEntry = data_.find(traitId);
+    if (traitEntry == data_.end()) {
+      return {};
+    }
+    trait::TraitProperties propertyKeys;
```

Comment:
[consider][minor] Micro-optimisation
```suggestion
    trait::TraitProperties propertyKeys;
    propertyKeys.reserve(traitEntry->second.size());    
```

Reply:
Added [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/595/commits/c21a05f065bd214e0de97e4bafab4be79a5e4be1).